libs/SalesforceSDKCore/SalesforceSDKCoreTests/SalesforceRestAPITests.m

```diff
@@ -587,6 +604,133 @@ - (void)testRetrieveError {
+     XCTAssertEqual([results[0][STATUS_CODE] intValue], 201, @"Wrong status for first request");
+     XCTAssertEqual([results[1][STATUS_CODE] intValue], 201, @"Wrong status for second request");
+     XCTAssertEqual([results[2][STATUS_CODE] intValue], 200, @"Wrong status for third request");
+     XCTAssertEqual([results[3][STATUS_CODE] intValue], 200, @"Wrong status for fourth request");
+
+     // Queries should have returned ids of newly created account and contact
+     NSString* accountId = ((NSDictionary *) results[0][RESULT])[LID];
+     NSString* contactId = ((NSDictionary *) results[1][RESULT])[LID];
+     NSString* idFromFirstQuery = ((NSDictionary *) results[2][RESULT])[RECORDS][0][ID];
+     NSString* idFromSecondQuery = ((NSDictionary *) results[3][RESULT])[RECORDS][0][ID];
+     XCTAssertEqualObjects(accountId, idFromFirstQuery, @"Account id not returned by query");
+     XCTAssertEqualObjects(contactId, idFromSecondQuery, @"Contact id not returned by query");
+ }
+
+ // Test for composite request
+ // Run a composite request that:
+ // - creates an account,
+ // - creates a contact (with newly created account as parent),
+ // - run a query that should return newly created account and contact
+ - (void) testCompositeRequest {
+
+     NSDictionary *fields;
+
+     // Create account
+     NSString *accountName = [self generateRecordName];
+     fields = @{NAME: accountName};
+
+     SFRestRequest *createAccountRequest = [[SFRestAPI sharedInstance]
+             requestForCreateWithObjectType:ACCOUNT
+                                     fields:fields
+     ];
+
+     // Create contact
+     NSString *contactName = [self generateRecordName];
+     fields = @{LAST_NAME: contactName, ACCOUNT_ID: @"@{refAccount.id}"};
+
+     SFRestRequest *createContactRequest = [[SFRestAPI sharedInstance]
+             requestForCreateWithObjectType:CONTACT
+                                     fields:fields
+     ];
+
+     // Query for account and contact
+     SFRestRequest *queryForContact = [[SFRestAPI sharedInstance]
+             requestForQuery:[NSString stringWithFormat:@"select Id, AccountId from Contact where LastName = '%@'", contactName]
+     ];
+
+     // Build composite request
+     SFRestRequest *batchRequest = [[SFRestAPI sharedInstance]
+             compositeRequest:@[createAccountRequest, createContactRequest, queryForContact]
+                       refIds:@[@"refAccount", @"refContact", @"refQuery"]
+                    allOrNone:YES
+     ];
+
+     // Send request
+     SFNativeRestRequestListener *listener = [self sendSyncRequest:batchRequest];
+
+     // Checking response
+     NSDictionary * response = listener.dataResponse;
+     NSArray<NSDictionary *>* results = response[COMPOSITE_RESPONSE];
+     XCTAssertEqual(3, results.count, "Wrong number of results");
+     XCTAssertEqual([results[0][HTTP_STATUS_CODE] intValue], 201, @"Wrong status for first request");
```

Comment:
The status code for composite response is in a field called "httpStatusCode" while the status code for batch response is in a field called "statusCode".